src/compiler/tsc.ts

```diff
@@ -415,7 +415,8 @@ namespace ts {
 
             const compileResult = compile(rootFileNames, compilerOptions, compilerHost);
 
-            if (!compilerOptions.watch) {
+            // Firefox has Object.prototype.watch
+            if (!compilerOptions.watch || !compilerOptions.hasOwnProperty("watch")) {

```

Comment:
can you please: 
- add a function `isWatchSet` (or something similar) that will contain this check
- replace all places where we currently have `compilerOptions.watch || !compilerOptions.hasOwnProperty("watch")` with the call to this function
- add a comment to `compilerOptions.watch` why it should not be checked directly and `isWatchSet` should be used instead